File: pharmcrew-admin/src/main/java/kr/ant/kpa/pharmcrew/service/PtaxService.java
package kr.ant.kpa.pharmcrew.service;

import kr.ant.kpa.pharmcrew.resp.PcResp;
import kr.ant.kpa.pharmcrew.resp.ptax.PtaxnoticeListResp;
import kr.ant.kpa.pharmcrew.resp.ptax.PtaxnoticeResp;
import kr.ant.kpa.pharmcrew.resp.ptax.PtaxqnaListResp;
import kr.ant.kpa.pharmcrew.resp.ptax.PtaxqnaResp;
import kr.ant.kpa.pharmcrew.validation.exception.FailSaveFileException;
import kr.ant.kpa.pharmcrew.validation.exception.gen.NotDeletablePtaxnoticeException;
import org.springframework.web.multipart.MultipartFile;

public interface PtaxService {
    PtaxqnaListResp qnaList(String regDtStartDate, String regDtEndDate, String target, String pharmName, String answerYn, String keyword, Integer page, Integer cntPerPage);
    PtaxqnaResp qna(Long qnaId);
    PcResp answerQna(Long qnaId, String answerTitle, String answerBody, Long[] delAttachFileIds, MultipartFile[] attachFiles, long answerId) throws FailSaveFileException;

    PcResp registerNotice(String sendType, String target, String targetPharmName, Long targetMemberId, String openYn, String reserveDate, String reserveHour, String reserveMinute, String osType, String title, String body, MultipartFile[] attachFiles, long regId) throws FailSaveFileException;
    PtaxnoticeListResp noticeList(String regDtStartDate, String regDtEndDate, String target, String targetPharmName, String sendType, String keyword, Integer page, Integer cntPerPage);
    PtaxnoticeResp notice(Long noticeId);
    PcResp updateNotice(Long noticeId, String sendType, String target, String targetPharmName, Long targetMemberId, String openYn, String reserveDate, String reserveHour, String reserveMinute, String osType, String title, String body, Long[] delAttachFileIds, MultipartFile[] attachFiles, long modId) throws FailSaveFileException;
    PcResp deleteNotice(Long noticeId) throws NotDeletablePtaxnoticeException;
}

File: pharmcrew-admin-ng/src/app/domain/ptax/ptax.service.ts
import { Injectable } from '@angular/core';
import { HttpService } from 'src/app/http.service';
import { PtaxqnaListReq } from './ptaxqna-list-req';
import { PtaxqnaListResp } from './ptaxqna-list-resp';
import { PtaxqnaResp } from './ptaxqna-resp';
import { AnswerPtaxqnaReq } from './answer-ptaxqna-req';
import { PcResp } from 'src/app/pc-resp';
import { PtaxnoticeListReq } from './ptaxnotice-list-req';
import { PtaxnoticeListResp } from './ptaxnotice-list-resp';
import { RegisterPtaxnoticeReq } from './register-ptaxnotice-req';
import { PtaxnoticeResp } from './ptaxnotice-resp';
import { UpdatePtaxnoticeReq } from './update-ptaxnotice-req';

@Injectable({
  providedIn: 'root'
})
export class PtaxService extends HttpService {
  // QNA
  qnaList(req: PtaxqnaListReq) {
    return this.get<PtaxqnaListResp>('/ptaxqnas', req);
  }
  qna(qnaId: number) {
    return this.get<PtaxqnaResp>(`/ptaxqnas/${qnaId}`);
  }
  answerQna(qnaId: number, req: AnswerPtaxqnaReq) {
    return this.postMultipart<PcResp>(`/ptaxqnas/${qnaId}`, req);
  }
  // 공지
  registerNotice(req: RegisterPtaxnoticeReq) {
    return this.postMultipart<PcResp>('/ptaxnotices', req);
  }
  noticeList(req: PtaxnoticeListReq) {
    return this.get<PtaxnoticeListResp>('/ptaxnotices', req);
  }
  notice(noticeId: number) {
    return this.get<PtaxnoticeResp>(`/ptaxnotices/${noticeId}`);
  }
  updateNotice(noticeId: number, req: UpdatePtaxnoticeReq) {
    return this.postMultipart<PcResp>(`/ptaxnotices/${noticeId}`, req);
  }
  deleteNotice(noticeId: number) {
    return this.delete<PcResp>(`/ptaxnotices/${noticeId}`);
  }
}
